pkg/kev/services_test.go

```diff
@@ -141,6 +143,190 @@ var _ = Describe("ServiceConfig", func() {
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadReadinessProbeTCPPort))
+					})
+
+					It("succeeds when port is provided", func() {
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeTCPPort] = "8080"
+
+						err := serviceConfig.validate()
+						Expect(err).To(Succeed())
+					})
+				})
+				Context("liveness", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeTCP.String()
+					})
+
+					It("fails when tcp probe has missing port", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadLivenessProbeTCPPort))
+					})
+
+					It("succeeds when port is provided", func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeTCPPort] = "8080"
+
+						err := serviceConfig.validate()
+						Expect(err).To(Succeed())
+					})
+				})
+			})
+
+			Context("command probe validation", func() {
+				Context("rediness probess", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeType] = kubernetes.ProbeTypeCommand.String()
+					})
+
+					It("fails when command probe has missing properties", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadReadinessProbeCommand))
+					})
+				})
+
+				Context("liveness probe", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeCommand.String()
+					})
+
+					It("fails when command probe has missing properties", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadLivenessProbeCommand))
+					})
+				})
+			})
+
+			Context("liveness probe type enum validation", func() {
+				BeforeEach(func() {
+					serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
+				})
+
+				Context("readiness", func() {
+					It("fails when type is misspelled", func() {
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeType] = "wrong-value"
+
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadReadinessProbeType))
+						Expect(err.Error()).To(ContainSubstring("none"))
+					})
+
+					It("succeeds if set to none", func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
```

Comment:
I guess you meant `config.LabelWorkloadLivenessProbeType` -> `config.LabelWorkloadReadinessProbeType`

Reply:
this can actually be removed because it's set on the BeforeEach, good catch